duktape/src/main/java/com/squareup/duktape/Duktape.java

```diff
@@ -125,8 +125,11 @@ private static void checkSignatureSupported(Method method) {
 
   /** Returns true if we support {@code: type} in calls from JavaScript. */
   private static boolean isSupportedType(Class<?> type) {
-    // TODO: support some more types.
-    return void.class.equals(type) || String.class.equals(type);
+    return void.class.equals(type)
+        || boolean.class.equals(type)
+        || int.class.equals(type)
+        || double.class.equals(type)
```

Comment:
Yeah. If a number can be represented exactly as a Java double, the conversion is lossless. Otherwise it’s lossy. The easiest way to check is to round trip cast & compare.